File: range-sum.js
// Time, Space -  O(N), O(N)
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} L
 * @param {number} R
 * @return {number}
 */
var rangeSumBST = function(root, L, R) {
    if (!root) {
        return 0
    }
    
    const { val, left, right } = root
    return (val >= L && val <= R ? val : 0) + rangeSumBST(left, L, R) + rangeSumBST(right, L, R)  
};

File: rangeSum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        sum1 = 0
        def preorder(root):
            if(root==None):
                return
            if(root.val>=low and root.val<=high):
                nonlocal sum1
                sum1+=root.val
                preorder(root.left)
                preorder(root.right)
            elif(root.val<low):
                preorder(root.right)
            else:
                preorder(root.left)
        
        preorder(root)
        return sum1
